### DISCUSSION TITLE
2FA with SSH

### DISCUSSION THREAD

**Discussion by dgalligani** [RobThree/TwoFactorAuth#67] [Q&A]
Hello Rob,
many thanks for your work. I'm David, a sysadmin at a Scientific Investigation Center in Spain. I was wondering if I can use your code to implement 2FA with ssh instead of using Google's.

**Comment by RobThree** [comment]
You probably can, but that would require making a [PAM module](https://en.wikipedia.org/wiki/Pluggable_authentication_module) and I have no idea or experience with this. Sorry.

Out of curiosity: what's wrong with Google's? Why wouldn't you use it?

**Comment by willpower232** [comment]
I'd not recommend attempting this as [Googles PAM module](https://github.com/google/google-authenticator-libpam) appears to be entirely self dependent, if you wrote a PAM module that used this PHP code to validate it and something went wrong with the PHP in your server, you could end up in a situation where you are not able to log in to your server.
